Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -719,4 +719,43 @@ def test_services
   ensure
     FileUtils.rm_rf HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-services"
   end
+
+  def test_link
+    assert_match "This command requires a keg argument", cmd_fail("link")
+
+    setup_test_formula "testball1"
+    cmd("install", "testball1")
+    cmd("link", "testball1")
+
+    cmd("unlink", "testball1")
+    assert_match "Would link", cmd("link", "--dry-run", "testball1")
+    assert_match "Would remove",
+      cmd("link", "--dry-run", "--overwrite", "testball1")
+    assert_match "Linking", cmd("link", "testball1")
+
+    setup_test_formula "testball2", <<-EOS.undent
+      keg_only "just because"
+    EOS
+    cmd("install", "testball2")
+    assert_match "testball2 is keg-only", cmd("link", "testball2")
+  ensure
+    FileUtils.rm_rf Pathname.glob HOMEBREW_CELLAR/"testball*"
+    FileUtils.rm_rf Pathname.glob HOMEBREW_CACHE/"testball*"
+    FileUtils.rm_rf Pathname.glob HOMEBREW_LOCK_DIR/"testball*"
```

Comment:
I'm wondering if you can just do e.g. `FileUtils.rm_rf HOMEBREW_LOCK_DIR` and skip the glob?